hbase-server/src/test/java/org/apache/hadoop/hbase/ipc/AbstractTestIPC.java

```diff
@@ -363,6 +365,101 @@ public void testAsyncEcho() throws IOException {
     }
   }
 
+  /**
+   * Tests the happy path for various request fan out values using a simple RPC hedged across
+   * a mix of running and failing servers.
+   */
+  @Test
+  public void testHedgedAsyncEcho() throws Exception {
+    List<RpcServer> rpcServers = new ArrayList<>();
+    List<InetSocketAddress> addresses = new ArrayList<>();
+    // Create a mix of running and failing servers.
+    final int numRunningServers = 5;
+    final int numFailingServers = 3;
```

Comment:
If this is happy-path testing, why are there any failing servers at all? When I think "happy path", I would expect a test with all running servers, and assertions that (1) the correct value is received every time and (2) no more than `fanOut` RPC calls are ever issued.